cmd/server-startup-msg.go

```diff
@@ -79,39 +80,41 @@ func printStartupMessage(apiEndpoints []string, err error) {
 	}
 }
 
-// Returns true if input is not IPv4, false if it is.
-func isNotIPv4(host string) bool {
+// Returns true if input is IPv6
+func isIPv6(host string) bool {
 	h, _, err := net.SplitHostPort(host)
 	if err != nil {
 		h = host
 	}
 	ip := net.ParseIP(h)
-	ok := ip.To4() != nil // This is always true of IP is IPv4
-
-	// Returns true if input is not IPv4.
-	return !ok
+	return ip.To16() != nil && ip.To4() == nil
```

Comment:
ip.To16() returning non-nil doesn't imply that it is an IPV6 address.
> a 16-byte slice can still be an IPv4 address. 
ref: https://pkg.go.dev/net#IP

Reply:
that is why To4() needs to be nil @krisis